R/xrf.R

```diff
@@ -2,101 +2,170 @@
 ## functions for preconditions on user input
 #############################################
 
-condition_xgb_control <- function(family, xgb_control, data, response_var, prefit_xgb) {
+condition_xgb_control <- function(
+  family,
+  xgb_control,
+  data,
+  response_var,
+  prefit_xgb,
+  call = rlang::caller_env()
+) {
   # this is a duplicated but necessary check
   if (!(response_var %in% colnames(data))) {
-    stop(paste0('Response variable "', response_var, '" not present in supplied data'))
+    cli::cli_abort(
+      "Response variable {.var {response_var}} not present in supplied data",
+      call = call
+    )
   }
 
   data_mutated <- data
 
-  if (family == 'multinomial' && is.null(xgb_control$num_class) && is.null(prefit_xgb)) {
+  if (
+    family == 'multinomial' &&
+      is.null(xgb_control$num_class) &&
+      is.null(prefit_xgb)
+  ) {
     n_classes <- n_distinct(data[[response_var]])
-    warning(paste0('Detected ', as.character(n_classes), ' classes to set num_class xgb_control parameter'))
+    cli::cli_warn(
+      "Detected {n_classes} class{?es} to set num_class xgb_control parameter",
+      call = call
+    )
     xgb_control$num_class <- n_distinct(data[[response_var]])
   }
 
   # xgboost expects multinomial labels to be 0:num_class
-  if (family == 'multinomial' &&
-      (is.factor(data[[response_var]]) || is.character(data[[response_var]]))) {
+  if (
+    family == 'multinomial' &&
+      (is.factor(data[[response_var]]) || is.character(data[[response_var]]))
+  ) {
     integer_response <- as.integer(as.factor(data[[response_var]]))
     data_mutated[[response_var]] <- integer_response - min(integer_response)
-  }
-  else if (family == 'binomial' &&
-           is.factor(data[[response_var]]) || is.character(data[[response_var]])) {
+  } else if (
+    family == 'binomial' &&
+      is.factor(data[[response_var]]) ||
+      is.character(data[[response_var]])
+  ) {
     integer_response <- as.integer(as.factor(data[[response_var]]))
     data_mutated[[response_var]] <- integer_response - min(integer_response)
   }
 
-  list(xgb_control = xgb_control,
-       data = data_mutated)
+  list(xgb_control = xgb_control, data = data_mutated)
 }
 
-xrf_preconditions <- function(family, xgb_control, glm_control,
-                              data, response_var, prefit_xgb) {
+xrf_preconditions <- function(
+  family,
+  xgb_control,
+  glm_control,
+  data,
+  response_var,
+  prefit_xgb,
+  call = rlang::caller_env()
```

Comment:
In case this looks weird, the `call` argument helps the error message reference the function the user expects (e.g. `xrf()` instead of `xrf_preconditions()`).